### ISSUE TITLE
gradle: merge engine-tests back in to engine directory

### ISSUE THREAD

**Issue by keturn** [MovingBlocks/Terasology#3979]
The directory (and gradle) layout for engine and its tests is like this:

* /engine/src/main
* /engine-tests/src/main
* /engine-tests/src/test

instead of the standard

* /engine/src/main
* /engine/src/test

Using the standard layout would simplify the build configuration.

Dumping some [thoughts](https://discord.com/channels/270264625419911192/696830442065625118/713789711793258587) from Cervator here:

> they're kinda structured like tests. Maybe we should just move them into engine-tests , add a Gradle JavaExec to run them, and redirect the output to a file we could then probably freestyle graph in Jenkins
> does there need to be an `engine-tests` _anymore_ - many moons ago we needed that hack, but i think Gradle (and IntelliJ?) both got more advanced about how you could link together sourceSets / code modules / compile vs test
> i don't remember the details anymore but it was definitely something about being unable to have it unified because modules (maybe specifically module tests?) would get confused and things wouldn't compile / test in the right order
> maybe particularly about tests that needed the environment base classes, like HeadlessEnvironment
> worth noting is that class (and its friends IIRC) all live in the regular sourceSet inside engine-tests

**Comment by Cervator** [comment]
On a related note the `dev` sourceSet inside `engine` can be simplified as well. It was kept separate for ... I dunno :) Probably just didn't seem right to have that code in the engine (it doesn't do anything for the engine) _nor_ in tests since it isn't junit based or anything, although it _feels_ kinda close to the tests, and benchmarking is sort of a test category

I propose (and have just validated that it'll work - look for and run `EntitySystemBenchmark`) moving the `org.terasology.benchmarks` package into the `main` sourceSet in `engine-tests` + remove the special handling for the `dev` sourceSet in the engine, although maybe that'll then be weird after re-unifying it since it _does_ still not fit as engine code, nor regular test code 🤔 

Once upon a time I was thinking we should have a *test* sourceSet per type of test - unit, integration, benchmark, acceptance (MTE) but maybe that'd just re-complicate things. It would really organize what kind of test something is though.

**Comment by keturn** [comment]
I have found one reason for making a `-tests` project that builds an independent thing from the main library: code you want to be re-used by other tests.

If you make something like ModuleTestingEnvironment, it has dependencies that aren't otherwise part of your library's `implementation` dependencies. Your users don't want this stuff on their `implementation` classpath either, they want something they can use as a `test` dependency.

I think you can do that _and_ still keep the tests for the engine under `src/test` in the main project. The dependencies get a little bit twisty, but I think this works:
- (Engine) - [:testImplementation] -> (MTE)  // allows project's tests to use MTE
- (MTE) - [:implementation] -> (Engine)    // allows MTE to use engine APIs

A gradle project's `test` source set depends on its `main` source set anyway, it should work out.

**tl;dr:** do move `engine-tests/src/test/` to `engine/src/test`, [move MTE](https://github.com/Terasology/ModuleTestingEnvironment/issues/39) to `engine-tests/src/main`.

**Comment by jdrueckert** [comment]
> tl;dr: do move `engine-tests/src/test/` to `engine/src/test`, move MTE to `engine-tests/src/main`

@keturn If we move MTE to `engine-tests/src/main`, would it make sense to rename it while we're at it? If not, why? And if so, to what? `integrationTestEnvironment` or something?

**Comment by keturn** [comment]
Sure, it'd be a convenient time to rename it.

Relevant terms include:
- Integration: it sets up a complex environment in which we test the integration of various different things
- Extension: the primary use case for it is as a JUnit Jupiter Extension
- Engine: it provides a TerasologyEngine instance
  - ModuleEnvironment: _most_ of the time it will be configured with a particular ModuleEnvironment (unless/until we have integration tests that involve something like going through Advanced Game Setup)
- Headless: it runs things without rendering, as a headless server would. (But it would be handy to be able to switch on visible-mode sometimes.)
